resources/shellHelperFunctions.sh

```diff
@@ -178,6 +178,51 @@ set_npm_version () {
   npm version --allow-same-version --no-git-tag-version --no-commit-hooks "$VERSION_WITH_TAG"
 }
 
+#
+# Re-runs the specified command-line instruction up to 5 times should it fail, waiting a
+# random delay between each attempt.  No re-runs are attempted after successful commands.
+#
+# ### Usage
+#   try_multiple_times command [param1 param2...]
+#
+# ### Parameters
+#   1: $@         command-line arguments
+try_multiple_times ( ) {
+  _try_multiple_times 0 "$@"
+}
+
+# $1  The current retry count
+# $2+ (everything else) the command to retry should it fail
+_try_multiple_times ( ) {
+  local RETRY_MAX=5
+
+  # Configuration:  wait between 10 sec and 120 sec.
+
+  # in seconds.
+  local RETRY_MAX_WAIT_RANGE=111
+  # in seconds
+  local RETRY_MIN_WAIT=10
+
+  local retryCount=$1
+  shift
+
+  if (( "$retryCount" == "$RETRY_MAX" )); then
+    builder_die "Retry limit of $RETRY_MAX attempts reached."
+  fi
+
+  retryCount=$(( $retryCount + 1 ))
+
+  if (( $retryCount != 1 )); then
+    local wait_length=$(( RANDOM % RETRY_MAX_WAIT_RANGE + RETRY_MIN_WAIT ))
+    builder_echo "Delaying $wait_length seconds before attempt $retryCount: \`$@\`"
+    sleep $wait_length
+  fi
+
+  if ! "$@"; then
```

Comment:
```suggestion
  if ! "$@"; then
    builder_echo "Command failed with error $?"
```

If would be really helpful to capture the error code here and report it. (Need to test that the exit code is not already lost; if it is then we'll need a slightly more complicated solution).

Reply:
I should've checked it locally first:

```
[web/src/app/browser] Command failed with error 0
```